.eslintrc.js

```diff
@@ -22,78 +56,393 @@ module.exports = {
     jsdom: true,
     L: true,
   },
+  overrides: [
+    {
+      files: ['**/*.test.js', 'src/test/utils.js'],
+      rules: {
+        'max-nested-callbacks': 'off',
+        'no-undef': 'off',
+        'require-await': 'off',
+        'promise/catch-or-return': 'off', // Shall we prefer await?
+        'promise/always-return': 'off',
+        'promise/no-callback-in-promise': 'off',
+        'no-magic-numbers': 'off',
+        'no-constructor-return': 'off',
+        'testing-library/prefer-find-by': 'off', // Needs some manual work
+        'promise/avoid-new': 'off',
+        'max-len': 'off',
+      },
+    },
+    {
+      files: ['internals/**/*'],
+      rules: {
+        'no-console': 'warn',
+      },
+    },
+    {
+      files: ['cypress/**/*'],
+      rules: {
+        'promise/always-return': 'off',
+        'promise/catch-or-return': 'off',
+      },
+    },
+  ],
   rules: {
-    camelcase: 'off',
-    'arrow-parens': ['error', 'as-needed', { requireForBlockBody: false }],
+    // proposed rules
```

Comment:
Once this is merged, the rules aren't part of a proposal any more. Remove this comment?

Reply:
I would love to see most of those rules implemented, but they need manual intervention or to be discussed, I would prefer to keep them here, so I or we can pick up those rules gradually.